server/src/book-bundle.ts

```diff
@@ -618,6 +618,45 @@ export class BookBundle {
     }[item.type][change.type].bind(this)
     func(item.key)
   }
+
+  isDirectoryDeletion(change: FileEvent): boolean {
+    if (change.type !== FileChangeType.Deleted) {
+      return false
+    }
+    const deletedPath = URI.parse(change.uri).fsPath
+
+    // This assumes both collections and media dirs are flat
+    if ((deletedPath === this.collectionDirectory()) || (deletedPath === this.mediaDirectory())) {
+      return true
+    }
+
+    return (deletedPath.includes(this.moduleDirectory()) && !deletedPath.endsWith('index.cnxml'))
```

Comment:
If someone mistakenly drops a non-`index.cnxml` file in a module directory and then deletes it, will this cause the model to try to drop a module that doesn't exist (which I don't believe will error, but it will bust the cache)?

i.e. should we include the following in our tests?
```
assert(!bundle.isDirectoryDeletion({ type: FileChangeType.Deleted, uri: '/bundle/modules/m00001/random.txt' }))
```
I see you explicitly mentioned this is NOT a catch-all solution, though. This may be one of those corner cases we don't care about, since invalidating the cache for a single module isn't really a huge deal.

Reply:
Hmmm...good point. As you suspect, currently the test case would fail, `onModuleDeleted(moduleId)` would get called with `foo.txt` not impacting the list of modules but busting the cache. Maybe instead of testing `!deletedPath.endsWith('index.cnxml'))` I can test that end of the path is a valid module ID to address the test case + be safer in general.

Reply:
[Done](https://github.com/openstax/poet/pull/60/commits/7b5a17630eaa6e821564f8034418bf0ff945da38)